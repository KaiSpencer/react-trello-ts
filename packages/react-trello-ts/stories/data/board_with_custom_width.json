{
  "lanes": [
    {
      "id": "PLANNED",
      "title": "Planned Tasks",
      "label": "20/70",
      "style": {"width": 280},
      "cards": [
        {
          "id": "Milk",
          "title": "Buy milk",
          "label": "15 mins",
          "cardStyle": { "width": 270, "maxWidth": 270, "margin": "auto", "marginBottom": 5 },
          "description": "2 Gallons of milk at the Deli store"
        },
        {
          "id": "Plan2",
          "title": "Dispose Garbage",
          "label": "10 mins",
          "cardStyle": { "width": 270, "maxWidth": 270, "margin": "auto", "marginBottom": 5 },
          "description": "Sort out recyclable and waste as needed"
        },
        {
          "id": "Plan3",
          "title": "Write Blog",
          "label": "30 mins",
          "cardStyle": { "width": 270, "maxWidth": 270, "margin": "auto", "marginBottom": 5 },
          "description": "Can AI make memes?"
        },
        {
          "id": "Plan4",
          "title": "Pay Rent",
          "label": "5 mins",
          "cardStyle": { "width": 270, "maxWidth": 270, "margin": "auto", "marginBottom": 5 },
          "description": "Transfer to bank account"
        }
      ]
    },
    {
      "id": "WIP",
      "title": "Work In Progress",
      "label": "10/20",
      "style": {"width": 280},
      "cards": [
        {
          "id": "Wip1",
          "title": "Clean House",
          "label": "30 mins",
          "cardStyle": { "width": 270, "maxWidth": 270, "margin": "auto", "marginBottom": 5 },
          "description": "Soap wash and polish floor. Polish windows and doors. Scrap all broken glasses"
        }
      ]
    },
    {
      "id": "BLOCKED",
      "title": "Blocked",
      "label": "0/0",
      "style": {"width": 280},
      "cards": []
    },
    {
      "id": "COMPLETED",
      "title": "Completed",
      "style": {"width": 280},
      "label": "2/5",
      "cards": [
        {
          "id": "Completed1",
          "title": "Practice Meditation",
          "label": "15 mins",
          "cardStyle": { "width": 270, "maxWidth": 270, "margin": "auto", "marginBottom": 5 },
          "description": "Use Headspace app"
        },
        {
          "id": "Completed2",
          "title": "Maintain Daily Journal",
          "label": "15 mins",
          "cardStyle": { "width": 270, "maxWidth": 270, "margin": "auto", "marginBottom": 5 },
          "description": "Use Spreadsheet for now"
        }
      ]
    },
    {
      "id": "REPEAT",
      "title": "Repeat",
      "style": {"width": 280},
      "label": "1/1",
      "cards": [
        {
          "id": "Repeat1",
          "title": "Morning Jog",
          "label": "30 mins",
          "cardStyle": { "width": 270, "maxWidth": 270, "margin": "auto", "marginBottom": 5 },
          "description": "Track using fitbit"
        }
      ]
    },
    {
      "id": "ARCHIVED",
      "title": "Archived",
      "style": {"width": 280},
      "label": "1/1",
      "cards": [
        {
          "id": "Archived1",
          "title": "Go Trekking",
          "label": "300 mins",
          "cardStyle": { "width": 270, "maxWidth": 270, "margin": "auto", "marginBottom": 5 },
          "description": "Completed 10km on cycle"
        }
      ]
    },
    {
      "id": "ARCHIVED2",
      "title": "Archived2",
      "style": {"width": 280},
      "label": "1/1",
      "cards": [
        {
          "id": "Archived1",
          "title": "Go Trekking",
          "label": "300 mins",
          "cardStyle": { "width": 270, "maxWidth": 270, "margin": "auto", "marginBottom": 5 },
          "description": "Completed 10km on cycle"
        }
      ]
    },
    {
      "id": "ARCHIVED3",
      "title": "Archived3",
      "style": {"width": 280},
      "label": "1/1",
      "cards": [
        {
          "id": "Archived1",
          "title": "Go Trekking",
          "label": "300 mins",
          "cardStyle": { "width": 270, "maxWidth": 270, "margin": "auto", "marginBottom": 5 },
          "description": "Completed 10km on cycle"
        }
      ]
    }
  ]
}
