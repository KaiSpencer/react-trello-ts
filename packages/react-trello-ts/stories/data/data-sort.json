
{
  "lanes": [
    {
      "id": "SORTED_LANE",
      "title": "Sorted Lane",
      "label": "20/70",
      "cards": [
        {
          "id": "Card1",
          "title": "Buy milk",
          "label": "2017-12-01",
          "description": "2 Gallons of milk at the Deli store",
          "metadata": {
            "completedAt": "2017-12-01T10:00:00Z",
            "shortCode": "abc"
          }
        },
        {
          "id": "Card2",
          "title": "Dispose Garbage",
          "label": "2017-11-01",
          "description": "Sort out recyclable and waste as needed",
          "metadata": {
            "completedAt": "2017-11-01T10:00:00Z",
            "shortCode": "aaa"
          }
        },
        {
          "id": "Card3",
          "title": "Write Blog",
          "label": "2017-10-01",
          "description": "Can AI make memes?",
          "metadata": {
            "completedAt": "2017-10-01T10:00:00Z",
            "shortCode": "fa1"
          }
        },
        {
          "id": "Card4",
          "title": "Pay Rent",
          "label": "2017-09-01",
          "description": "Transfer to bank account",
          "metadata": {
            "completedAt": "2017-09-01T10:00:00Z",
            "shortCode": "ga2"
          }
        }
      ]
    }
  ]
}
