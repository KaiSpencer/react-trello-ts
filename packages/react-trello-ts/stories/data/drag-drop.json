{
  "lanes": [
    {
      "id": "PLANNED",
      "title": "Planned Tasks",
      "label": "20/70",
      "cards": [
        {
          "id": "Milk",
          "title": "Buy milk",
          "label": "15 mins",
          "description": "2 Gallons of milk at the Deli store"
        },
        {
          "id": "Plan2",
          "title": "Dispose Garbage",
          "label": "10 mins",
          "description": "Sort out recyclable and waste as needed"
        },
        {
          "id": "Plan3",
          "title": "Write Blog",
          "label": "30 mins",
          "description": "Can AI make memes?"
        },
        {
          "id": "Plan4",
          "title": "Pay Rent",
          "label": "5 mins",
          "description": "Transfer to bank account"
        }
      ]
    },
    {
      "id": "WIP",
      "title": "Work In Progress (Not Droppable)",
      "label": "10/20",
      "droppable": false,
      "cards": [
        {
          "id": "Wip1",
          "title": "Clean House",
          "label": "30 mins",
          "description": "Soap wash and polish floor. Polish windows and doors. Scrap all broken glasses"
        }
      ]
    },
    {
      "id": "COMPLETED",
      "title": "Completed (Droppable)",
      "label": "0/0",
      "style": {"width": 280},
      "cards": []
    }
  ]
}