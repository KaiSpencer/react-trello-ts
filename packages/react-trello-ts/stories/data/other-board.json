{
  "lanes": [
    {
      "id": "yesterday",
      "title": "Yesterday",
      "label": "20/70",
      "cards": [
        {
          "id": "Wip1",
          "title": "Clean House",
          "label": "30 mins",
          "description": "Soap wash and polish floor. Polish windows and doors. Scrap all broken glasses"
        }
      ]
    },
    {
      "id": "today",
      "title": "Today",
      "label": "10/20",
      "droppable": false,
      "cards": [
        {
          "id": "Milk",
          "title": "Buy milk",
          "label": "15 mins",
          "description": "2 Gallons of milk at the Deli store"
        },
        {
          "id": "Plan2",
          "title": "Dispose Garbage",
          "label": "10 mins",
          "description": "Sort out recyclable and waste as needed"
        },
        {
          "id": "Plan3",
          "title": "Write Blog",
          "label": "30 mins",
          "description": "Can AI make memes?"
        },
        {
          "id": "Plan4",
          "title": "Pay Rent",
          "label": "5 mins",
          "description": "Transfer to bank account"
        }
      ]
    },
    {
      "id": "tomorrow",
      "title": "Tomorrow",
      "label": "0/0",
      "cards": []
    }
  ]
}