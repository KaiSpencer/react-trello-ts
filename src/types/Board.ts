import {CSSProperties} from 'react'

export interface BoardData {
  lanes: Lane[]
}

export interface Lane {
  id: string
  title?: string
  label?: string
  style?: CSSProperties
  cards: Card[]
  currentPage?: number
  droppable?: boolean
  labelStyle?: CSSProperties
  cardStyle?: CSSProperties
  disallowAddingCard?: boolean
  [key: string]: any
}

export interface Card {
  id: string
  title?: string
  label?: string
  description?: string
  laneId?: string
  style?: CSSProperties
  draggable?: boolean
  [key: string]: any
}
