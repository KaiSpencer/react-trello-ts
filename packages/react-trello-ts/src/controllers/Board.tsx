import React, { createContext, FC, PropsWithChildren } from "react";
import classNames from "classnames";
import uuidv1 from "uuid/v1";
import { BoardContainer } from "./BoardContainer";
import * as DefaultComponets from "../components";
import { BoardData } from "../types/Board";
import { store } from "../store/store";

export const Board: FC<
	PropsWithChildren<{
		id?: string;
		className?: string;
		components: typeof DefaultComponets;
		data: BoardData;
		t?: any;
	}>
> = ({ data, children, className, components, id, t, ...rest }) => {
	const allClassNames = classNames("react-trello-board", className || "");

	return (
		<BoardContext.Provider value={store}>
			<components.GlobalStyle />
			<BoardContainer
				id={id || uuidv1()}
				data={data}
				components={components}
				t={t}
				className={allClassNames}
				{...rest}
			/>
		</BoardContext.Provider>
	);
};

export const BoardContext = createContext(store);
