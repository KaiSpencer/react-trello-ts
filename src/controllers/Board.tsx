import classNames from "classnames";
import React, { FC, PropsWithChildren, createContext } from "react";
import { v1 } from "uuid";
import * as DefaultComponets from "../components";
import { store } from "../store/store";
import { BoardData } from "../types/Board";
import { BoardContainer } from "./BoardContainer";

export const Board: FC<
	PropsWithChildren<{
		id?: string;
		className?: string;
		components: typeof DefaultComponets;
		data: BoardData;
		t?: any;
	}>
> = ({ data, children, className, components, id, t, ...rest }) => {
	const allClassNames = classNames("react-trello-board", className || "");

	return (
		<BoardContext.Provider value={store}>
			{/*
			// @ts-ignore */}
			<components.GlobalStyle />
			<BoardContainer
				id={id || v1()}
				data={data}
				components={components}
				t={t}
				className={allClassNames}
				{...rest}
			/>
		</BoardContext.Provider>
	);
};

export const BoardContext = createContext(store);
