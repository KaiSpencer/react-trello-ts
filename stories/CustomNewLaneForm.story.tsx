import React, { Component } from "react";

import Board from "../src";

import data from "./data/data-sort.json";
Board.displayName = "Board";

class NewLaneFormComponent extends Component {
	render() {
		const { onCancel, t } = this.props;
		const handleAdd = () => this.props.onAdd({ title: this.inputRef.value });
		const setInputRef = (ref) => (this.inputRef = ref);
		return (
			<div>
				<input ref={setInputRef} placeholder={t("placeholder.title")} />
				<button onClick={handleAdd}>{t("button.Add lane")}</button>
				<button onClick={onCancel}>{t("button.Cancel")}</button>
			</div>
		);
	}
}

export default {
	title: "Custom Components",
	component: () => (
		<Board
			editable={true}
			canAddLanes={true}
			components={{ NewLaneForm: NewLaneFormComponent }}
			data={data}
		/>
	),
};

export const NewLaneForm = {};
