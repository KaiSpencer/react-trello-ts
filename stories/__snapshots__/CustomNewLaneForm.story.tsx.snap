// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Components NewLaneForm smoke-test 1`] = `
<div draggable="false"
     class="sc-iGgWBj iSNfA"
>
  <div>
    <div class="smooth-dnd-container horizontal">
      <section draggable="false"
               class="sc-kAyceB dgWEjS react-trello-lane"
      >
        <header class="sc-gsFSXq sc-imWYAI imRIDz gLxuAH">
          <span class="sc-kpDqfm gDkBAH">
            Sorted Lane
          </span>
          <span class="sc-dAlyuH iHiWje">
            <span>
              20/70
            </span>
          </span>
        </header>
        <div class="sc-dhKdcB ikKiDc">
          <div class="smooth-dnd-container vertical">
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Card1"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    <textarea placeholder="title"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; height: 18px;"
                    >
                      Buy milk
                    </textarea>
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    <textarea placeholder="label"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; height: 36px;"
                    >
                      2017-12-01
                    </textarea>
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  <textarea placeholder="description"
                            autocomplete="off"
                            autocorrect="off"
                            autocapitalize="off"
                            spellcheck="false"
                            rows="1"
                            class="sc-fHjqPf bsHyvp"
                            style="overflow: hidden; overflow-wrap: break-word; height: 36px;"
                  >
                    2 Gallons of milk at the Deli store
                  </textarea>
                </div>
              </article>
            </div>
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Card2"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    <textarea placeholder="title"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; height: 18px;"
                    >
                      Dispose Garbage
                    </textarea>
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    <textarea placeholder="label"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; height: 36px;"
                    >
                      2017-11-01
                    </textarea>
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  <textarea placeholder="description"
                            autocomplete="off"
                            autocorrect="off"
                            autocapitalize="off"
                            spellcheck="false"
                            rows="1"
                            class="sc-fHjqPf bsHyvp"
                            style="overflow: hidden; overflow-wrap: break-word; height: 36px;"
                  >
                    Sort out recyclable and waste as needed
                  </textarea>
                </div>
              </article>
            </div>
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Card3"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    <textarea placeholder="title"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; height: 18px;"
                    >
                      Write Blog
                    </textarea>
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    <textarea placeholder="label"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; height: 36px;"
                    >
                      2017-10-01
                    </textarea>
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  <textarea placeholder="description"
                            autocomplete="off"
                            autocorrect="off"
                            autocapitalize="off"
                            spellcheck="false"
                            rows="1"
                            class="sc-fHjqPf bsHyvp"
                            style="overflow: hidden; overflow-wrap: break-word; height: 18px;"
                  >
                    Can AI make memes?
                  </textarea>
                </div>
              </article>
            </div>
            <div class="smooth-dnd-draggable-wrapper">
              <article data-id="Card4"
                       class="sc-jlZhew sc-cwHptR clGdKf fQNcVl react-trello-card"
              >
                <header class="sc-gsFSXq sc-jEACwC imRIDz jNaWkr">
                  <span draggable="true"
                        class="sc-kpDqfm sc-cPiKLX cFicSq cnDDQK"
                  >
                    <textarea placeholder="title"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; height: 18px;"
                    >
                      Pay Rent
                    </textarea>
                  </span>
                  <span class="sc-dAlyuH sc-dLMFU iHiWje dvkFIZ">
                    <textarea placeholder="label"
                              autocomplete="off"
                              autocorrect="off"
                              autocapitalize="off"
                              spellcheck="false"
                              rows="1"
                              class="sc-fHjqPf bsHyvp"
                              style="overflow: hidden; overflow-wrap: break-word; height: 36px;"
                    >
                      2017-09-01
                    </textarea>
                  </span>
                  <div class="sc-hmdomO gJuJJJ">
                    <button class="sc-jsJBEP djrAfI">
                      ✖
                    </button>
                  </div>
                </header>
                <div class="sc-eDPEul cRhiim">
                  <textarea placeholder="description"
                            autocomplete="off"
                            autocorrect="off"
                            autocapitalize="off"
                            spellcheck="false"
                            rows="1"
                            class="sc-fHjqPf bsHyvp"
                            style="overflow: hidden; overflow-wrap: break-word; height: 18px;"
                  >
                    Transfer to bank account
                  </textarea>
                </div>
              </article>
            </div>
          </div>
          <a class="sc-gFqAkR ecfzPu">
            Click to add card
          </a>
        </div>
      </section>
    </div>
  </div>
  <div class="smooth-dnd-container horizontal">
    <section class="sc-dAbbOL sc-feUZmu kUHSgc bMulzd">
      <button class="sc-fhzFiK duRxJO">
        + Add another lane
      </button>
    </section>
  </div>
</div>
`;
